apps/server/src/shared/domain/entity/submission.entity.ts

```diff
@@ -75,15 +75,101 @@ export class Submission extends BaseEntityWithTimestamps {
 		}
 	}
 
-	isGraded(): boolean {
-		const isGraded =
-			(typeof this.grade === 'number' && this.grade >= 0) ||
-			(typeof this.gradeComment === 'string' && this.gradeComment.length > 0) ||
-			(this.gradeFiles !== undefined && this.gradeFiles.length > 0);
+	private getCourseGroupStudentIds(): EntityId[] {
+		let courseGroupMemberIds: EntityId[] = [];
+
+		if (this.courseGroup) {
+			courseGroupMemberIds = this.courseGroup.getStudentIds();
+		}
+
+		return courseGroupMemberIds;
+	}
+
+	private getTeamMemberIds(): EntityId[] {
+		if (!this.teamMembers) {
+			throw new InternalServerErrorException(
+				'Submission.teamMembers is undefined. The submission need to be populated.'
+			);
+		}
+
+		const teamMemberObjectIds = this.teamMembers.getIdentifiers('_id');
+		const teamMemberIds = teamMemberObjectIds.map((id): string => id.toString());
+
+		return teamMemberIds;
+	}
+
+	private getGradeFileIds(): EntityId[] {
+		const gradeFilesObjectIds = this.gradeFiles.getIdentifiers('_id');
+		const gradeFilesIds = gradeFilesObjectIds.map((id): string => id.toString());
+
+		return gradeFilesIds;
+	}
+
+	private gradeExists(): boolean {
+		const gradeExists = typeof this.grade === 'number' && this.grade >= 0;
+
+		return gradeExists;
+	}
+
+	private gradeCommentExists(): boolean {
+		const gradeCommentExists = typeof this.gradeComment === 'string' && this.gradeComment.length > 0;
+
+		return gradeCommentExists;
+	}
+
+	private gradeFilesExists(): boolean {
+		const gradedFileIds = this.getGradeFileIds();
+		const gradeFilesExists = gradedFileIds.length > 0;
+
+		return gradeFilesExists;
+	}
+
+	public isSubmitted(): boolean {
+		// Always submitted for now, but can be changed in future.
+		const isSubmitted = true;
```

Comment:
It is need if we changed it or not. Before implicit the creating run into isSubmitted. Same situation in all test too. But it should exist explicit to say yes the creating === isSubmitted. If we want change it in future we can it, if not it is also required. 

For this concret case we want to change it with the next ticket.

Reply:
yes, we will need this for sure - there is an initiative to handle submission, but could be some time until then
Consider however to name the opposite, as in the task
 => isDraft() instead of isSubmitted() - where by default would be false

This also avoids naming like isSubmissionSubmitted() which is hard to read :) - would be !isDraft() for example

Reply:
@bischofmax @dyedwiper @SevenWaysDP the point from the daily.